pwr.md

```diff
@@ -0,0 +1,144 @@
+# pwr module
+
+This is a python module which allows an application to modify power attributes of a CPU. Manipulation can be done to the frequency of cores, 
+frequnecy of the uncore, frequency profiles can be set to achieve desired performance, as well as that capabilities of a CPU can be obtained and its stats monitered.  
+Configuration is done by first creating core and/or CPU object lists, these objects will store the characteristics of the physical processor.
+Modifying those objects and committing the changes enables the module to take the necessary steps to make those changes on the CPU hardware.
+
+## Initialization 
+Creation of the cpu and core object lists is done using the `get_cores()/get_cpus()` functions, which return a list of the respective objects.
+
+```python 
+import pwr
+
+cores = pwr.get_cores()
+cpus = pwr.get_cpus()
+```
+
+## Adjusting Power Configuration
+### Modifying
+Each cpu and core object have attributes which replicate the capabilities and stats of the physical cores and cpus.
+### cpu
+* `cpu_id`                  # CPU id number
+* `physical_id`             # physical CPU number
+* `core_list`               # list of core objects on this CPU
+* `sst_bf_enabled`          # base frequency enabled in BIOS
+* `sst_bf_configured`       # core configured to use base frequencies
+* `turbo_enabled`           # turbo enabled flag
+* `hwp_enabled`             # HWP enabled flag
+* `base_freq`               # base frequency
+* `all_core_turbo_freq`     # all core turbo frequency
+* `highest_freq`            # highest available frequency
+* `lowest_freq`             # lowest active frequency
+* `uncore_hw_max`           # max available uncore frequency
+* `uncore_hw_min`           # min available uncore frequency
+* `uncore_freq`             # current uncore frequency
+* `uncore_max_freq`         # max desired uncore frequency
+* `uncore_min_freq`         # min desired uncore frequency
+
+### core
+* `core_id`                 # core id number
+* `cpu`                     # this cores cpu object
+* `thread_siblings`         # list of thread siblings
+* `high_prioirty`           # high/normal priority
+* `base_freq`               # base frequency [2300Mhz]
+* `sst_bf_base_freq`        # priority based frequency [2100Mhz/2700Mhz]
+* `all_core_turbo_freq`     # all core turbo frequency [2800Mhz]
+* `highest_freq`            # highest frequency available [3900Mhz]
+* `lowest_freq`             # lowest active frequency [800Mhz]
+* `curr_freq`               # current core frequency
+* `min_freq`                # desired low frequency
+* `max_freq`                # desired high frequency
+* `epp`                     # energy performance preference
+
+The **core** object attributes which can be adjusted are `max_freq`, `min_freq` and `epp`, min and max set the desired core frequency at which an application wants the core to operate within. 
```

Comment:
Core?

You already have a doc string above, why duplicate it here? Also, i would reword it something along these lines:
```
Most of the **Core** object attributes are constant and cannot be changed. The only attributes that can be written to by the user, are `min_freq`, `max_freq` and `epp`.
```
The key point we want to get across is that most of the attributes aren't writable.

Reply:
Where is the duplication?
I would say we need to give the user more info than just "here are the values".

Reply:
"min and max set the desired core frequency at which an application wants the core to operate within" is already touched upon by the doc strings. We could perhaps expand on this a bit and provide some general overview of the concepts involved in using this library (such as how to use min/max freqs, epp, etc.), but here is not the place to do it, it should be in a separate section.

Reply:
when you say general overview of the concepts, what else would you say?
I did originally was going to make another section but there wasn't much to it so I put it here.

Reply:
For example, something like this:

# Give a general overview - e.g. the machine may have one or more CPU's (corresponding to sockets), each CPU has multiple physical cores, and each physical core may optionally have multiple threads (multiple logical cores sitting on the same physical core)
# Explain the CPU part - e.g. CPU has uncore which can be scaled within the limits set by min/max, and it is the place to go when one needs to figure out power consumption (btw this is missing from your API doc - i forgot to add it to readme, apologies for that!)
# Explain the core part - e.g. core can scale its frequency up or down, but only if all thread siblings agree on min/max frequencies (explain how it works), cores can also use SST-BF to set up their frequencies according to SST-BF configuration, and cores can take advantage of SST-CP by setting a specific EPP policy.

Note that none of the above requires referencing specific object attributes - these are general concepts. I think that's plenty to write about to make it an entire section :)